src/AccordionToggle.js

```diff
@@ -0,0 +1,72 @@
+import classNames from 'classnames';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { elementType } from 'prop-types-extra';
+
+import { createBootstrapComponent } from './ThemeProvider';
+import AccordionContext from './AccordionContext';
+
+class AccordionToggle extends React.Component {
+  static propTypes = {
+    /**
+     * Set a custom element for this component.
+     */
+    as: elementType,
+
+    /** @default 'accordion-toggler' */
+    bsPrefix: PropTypes.string,
+    /**
+     * A key that corresponds to the collapse component that gets triggered
+     * when this has been clicked.
+     */
+    eventKey: PropTypes.string.isRequired,
+
+    /**
+     * A callback function to perform additional operations with this toggler
+     * gets triggered.
+     */
+    onClick: PropTypes.func,
+  };
+
+  static defaultProps = {
+    as: 'button',
+  };
+
+  handleClick = () => {
+    const handleClick = this.accordionContext;
```

Comment:
maybe call this `accordionContextOnClick` or something. or just bind `onClick` inside `render`?